plugins/hls-semantic-tokens-plugin/test/testdata/TOperator.hs

```diff
@@ -0,0 +1,12 @@
+module TOperator where
+
+-- imported operator
+go f x = f $ x
+-- operator defined in local module
+($$$$) = b
+x = 1 $$$$ 2
+data a :+: b = Add a b
+-- type take precedence over operator
```

Comment:
I actually think people might prefer to see type operators highlighted differently? In "normal" Haskell syntax highlighting they are!

Reply:
Yeah, I've been thinking about this, may be it better for us to mark all the infix thing as operator.

Reply:
I feel reasonably sure that we want "operator" to take precedence over "type". I'm still unsure about "operator" vs "class method"

Reply:
me neither, maybe we can make a pool or something to collect people's opinion on this.

Reply:
Yep, lets stick to the simplest way for now. Make the infix on top of all.
We can sort things out in configuration PR that switching to the `[HsSemanticTokenType]` .